lib/src/lifecycle_module.dart

```diff
@@ -52,12 +53,18 @@ enum LifecycleState {
 /// Intended to be extended by most base module classes in order to provide a
 /// unified lifecycle API.
 abstract class LifecycleModule extends SimpleModule with Disposable {
+  static int _nextId = 0;
+
+  int _instanceId = _nextId++;
+
   List<LifecycleModule> _childModules = [];
   Logger _logger;
   String _name;
```

Comment:
I think I'd prefer renaming this to `_defaultName` so that the check you do to see if the module author has overridden the name or not makes more sense:

```dart
if (name == _defaultName) { ... }
```